jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleProjectPropertiesTest.java

```diff
@@ -589,17 +598,51 @@ private BuildConfiguration setupBuildConfiguration(
             .setAppRoot(AbsoluteUnixPath.get(appRoot))
             .setModificationTimeProvider((ignored1, ignored2) -> SAMPLE_FILE_MODIFICATION_TIME);
     JibContainerBuilder jibContainerBuilder =
-        new GradleProjectProperties(mockProject, mockLogger)
+        new GradleProjectProperties(mockProject, mockLogger, mocktempDirectoryProvider)
             .createJibContainerBuilder(javaContainerBuilder, containerizingMode);
     return JibContainerBuilderTestHelper.toBuildConfiguration(
         jibContainerBuilder,
         Containerizer.to(RegistryImage.named("to"))
             .setExecutorService(MoreExecutors.newDirectExecutorService()));
   }
 
-  private void setUpWarProject(Path webAppDirectory) {
-    Mockito.when(mockProject.getBuildDir()).thenReturn(webAppDirectory.toFile());
-    Mockito.when(mockTaskContainer.named("war")).thenReturn(warTaskProvider);
+  private Path setUpWarProject(Path webAppDirectory) throws IOException {
```

Comment:
This doesn't look right. I'll get back to you once you resolve other issues.

Reply:
Returning the `unzipTarget` so that the assertion would know where to find the expanded war contents.